var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -113,11 +113,14 @@ def mpi_dependent_options(self):
                 raise RuntimeError('\n'.join(errors))
         else:
             compiler_opts = [
-                '--with-cpp=cpp',
-                '--with-cxxcpp=cpp',
                 '--with-mpi=1',
                 '--with-mpi-dir=%s' % self.spec['mpi'].prefix,
             ]
+        if sys.platform != "darwin":
```

Comment:
I'm running OSX El Capitan and have spack at `462a4a1`, building with `clang + gfortran`. I've hit this bug and have also found that I've had to remove the 
```
'--with-cpp=cpp',
'--with-cxxcpp=cpp',
```
lines to get `PETSc` to build. I'm not sure why I have troubles building this while @davydden doesn't. Apart from his upgraded system, we have pretty much the same setup.